src/views/user/AnonymousLogin.js

```diff
@@ -1,54 +1,44 @@
 import React, { useContext, useState } from "react";
-import {useHistory} from "react-router-dom";
-import {Modal, Form, Image, Header, Message } from "semantic-ui-react";
-import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom";
 
 import { UserContext } from "contexts/UserContext";
-import OtherLoginOptions from "components/user/LoginRedirections";
+import CredentialsLayout from "components/layout/CredentialsLayout";
+import GuestForm from "components/user/forms/GuestForm";
 import handleAuthorization from "utils/user/handleAuthorization";
 
-
 const AnonymousLoginPage = () => {
   const [nickname, setNickname] = useState("");
   const [error, setError] = useState(null);
   const [isLoading, setIsLoading] = useState(false);
 
-  const { t } = useTranslation();
   const { logInAnonymously } = useContext(UserContext);
   const history = useHistory();
 
   const formValid = nickname.length > 0;
-  const handleLoginFunc = handleAuthorization(() => logInAnonymously(nickname), setError, setIsLoading, history);
+  const handleLoginFunc = handleAuthorization(
+    () => logInAnonymously(nickname),
+    setError,
+    setIsLoading,
+    history
+  );
 
   return (
-    <Modal open={true}>
-      <Modal.Header>{t("guest.prompt")}</Modal.Header>
-      <Modal.Content image>
-        <Image wrapped size="medium" src="/images/game-hugo.png" />
-        <Modal.Description>
-          <Header>{t("login.actions.guest_login")}</Header>
-          <Form onSubmit={handleLoginFunc} loading={isLoading} error={!!error}>
-            <Message error content={error} />
-            <Form.Input
-              autoFocus
-              maxLength="24"
-              placeholder={t("register.form.nickname")}
-              name={t("register.form.nickname")}
-              value={nickname}
-              onChange={(_, { value }) => setNickname(value)}
-            />
-            <Form.Group>
-              <Form.Button
-                disabled={!formValid}
-                color="green"
-                content={t("login.actions.guest_login")}
-              />
-            </Form.Group>
-          </Form>
-          <OtherLoginOptions setError={setError} setLoading={setIsLoading}/>
-        </Modal.Description>
-      </Modal.Content>
-    </Modal>
+    <CredentialsLayout
+      withLoginOptions
+      action="guest"
+      {...{
+        setError,
+        setIsLoading,
+        handleLoginFunc,
+        isLoading,
+        error,
+        nickname,
+        setNickname,
+        formValid,
+      }}
+    >
+      <GuestForm />
```

Comment:
Is it ok this way (passing all props to layout and layout passing it down) or should I change it so that Form explicitly receives whatever it needs here?

Reply:
So something like:
```
<CredentialsLayout withLoginOptions action="guest" setError={setError} setIsLoading={setIsLoading}>
  <GuestForm {...{setError, setIsLoading, handleLoginFunc, isLoading, error, nickname, setNickname, formValid}} />
</CredentialsLayout>
```

Reply:
This makes sense, because why should Layout care about `nickname` or `loginFunc`? 🤔

Reply:
I've changed it. It didn't make sense in the previous form